ndk/src/events/index.ts

```diff
@@ -310,6 +311,29 @@ export class NDKEvent extends EventEmitter {
     public encode = encode.bind(this);
     public encrypt = encrypt.bind(this);
     public decrypt = decrypt.bind(this);
+    public giftWrap = giftWrap.bind(this);
+    public giftUnwrap = giftUnwrap.bind(this);
+
+    /**
+     * Shorthand method for Nip-17 Private Direct Messages encryption using Nip-44 and Nip-29 gift wraps.
+     * 
+     * @param recipient user which receives the message
+     * @param signer optional signer to use for decryption of the events, only needed to overwrite the event's ndk signer
+     * @returns Promise containing the encrypted event if successful, otherwise an error
+     */
+    public async encryptNip17(recipient: NDKUser, signer?: NDKSigner): Promise<NDKEvent> {
```

Comment:
I'm not sure why these methods are helpful. They're one line each... what am I missing?

Reply:
giftWrap is a general method allowing gift-wrap construction with nip-04 or nip-44 and some other parameters. With this method I wanted to give developers a shorthand for the more common case, nip-17 private messages. But if you think it is not worth it or it doesn't fit well in ndk, I can remove it.